kato/kato-core/src/main/groovy/com/netflix/spinnaker/kato/helpers/OperationPoller.groovy

```diff
@@ -0,0 +1,93 @@
+package com.netflix.spinnaker.kato.helpers
+
+import com.netflix.spinnaker.kato.data.task.Task
+import com.netflix.spinnaker.kato.exception.OperationTimedOutException
+
+/**
+ * A poller with an upper time limit combined with a Fibonacci-based backoff.
+ * Let's you wrap the operation in a Groovy closure and the "if complete" operation in another.
+ * The "if complete" closure is fed the results of the first.
+ */
+class OperationPoller {
+
+  int asyncOperationTimeoutSecondsDefault
+  int asyncOperationMaxPollingIntervalSeconds
+
+  ThreadSleeper threadSleeper = new ThreadSleeper()
+
+  /**
+   * Wrap an operational closure with a back off algorithm to check until completed.
+   *
+   * @param operation - a closure to perform an operation and return a testable value
+   * @param ifDone - a closure that receives the operation's results, and returns true/false if completed
+   * @param timeoutSeconds
+   * @param task
+   * @param resourceString
+   * @param basePhase
+   * @return results of operation
+   */
+  Object waitForOperation(Closure operation, Closure ifDone,
+                          Long timeoutSeconds, Task task, String resourceString, String basePhase) {
+    return handleFinishedAsyncOperation(
+        pollOperation(operation, ifDone, getTimeout(timeoutSeconds)), task, resourceString, basePhase)
+  }
+
+  private long getTimeout(Long timeoutSeconds) {
+    // Note that we cannot use an Elvis operator here because we might have a timeoutSeconds value of
+    // zero. In that case, we still want to pass that value. So we use null comparison here instead.
+    Math.max(timeoutSeconds != null ? timeoutSeconds : asyncOperationTimeoutSecondsDefault, 0)
+  }
+
+  private static handleFinishedAsyncOperation(Object operation, Task task, String resourceString, String basePhase) {
+    if (!operation) {
+      String errorMsg = "Operation on $resourceString timed out."
+      task.updateStatus basePhase, errorMsg
+      throw new OperationTimedOutException(errorMsg)
+    }
+
+    task.updateStatus basePhase, "Done operating on $resourceString."
+
+    operation
+  }
+
+  private Object pollOperation(Closure operation, Closure ifDone, long timeoutSeconds) {
+    int totalTimePollingSeconds = 0
+    boolean timeoutExceeded = false
+
+    // Fibonacci backoff in seconds, up to googleConfigurationProperties.asyncOperationMaxPollingIntervalSeconds interval.
```

Comment:
Old comment. It's configured by constructor injection, and each bean can tap their own prop file.